### PR TITLE
lammps: add support for latte package

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -60,8 +61,14 @@ def url_for_version(self, version):
...
     depends_on('blas', when='+user-atc')
     depends_on('lapack', when='+user-atc')
+    depends_on('latte', when='+latte')
+    depends_on('blas', when='+latte')
+    depends_on('lapack', when='+latte')
+
+    conflicts('+latte', when='@:20170921')
 
     patch("lib.patch", when="@20170901")
+    patch("660.patch", when="@develop")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/lammps/package.py]: I will drop this when lammps/lammps#660 is merged.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 2]
```diff
@@ -37,13 +37,14 @@ class Lammps(CMakePackage):
...
     version('20170901', '767e7f07289663f033474dfe974974e7')
+    version('develop', git='https://github.com/lammps/lammps', branch='master')
 
     def url_for_version(self, version):
         vdate = dt.datetime.strptime(str(version), "%Y%m%d")
         return "https://github.com/lammps/lammps/archive/patch_{0}.tar.gz".format(
             vdate.strftime("%d%b%Y").lstrip('0'))
 
-    supported_packages = ['voronoi', 'rigid', 'user-nc-dump', 'kspace',
+    supported_packages = ['voronoi', 'rigid', 'user-nc-dump', 'kspace', 'latte',
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/lammps/package.py]: flake issue here (line is too long)

**Comment by junghans** [comment]
@davydden @scheibelp This is now ready to merge, in the meanwhile version 20170922 got release, so I added that one, too.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 3]
```diff
@@ -37,14 +37,16 @@ class Lammps(CMakePackage):
     url      = "https://github.com/lammps/lammps/archive/patch_1Sep2017.tar.gz"
 
     version('20170901', '767e7f07289663f033474dfe974974e7')
+    version('20170922', '4306071f919ec7e759bda195c26cfd9a')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/lammps/package.py]: we usually put them in descending order (sometimes with exception of `develop` being at the very bottom).

Comment (junghans) [reply var/spack/repos/builtin/packages/lammps/package.py]: Fixed!

**Comment by junghans** [comment]
Ping @scheibelp

**Pull request merged**